cloudapi/logs.go

```diff
@@ -150,8 +176,16 @@ func (c *Config) StreamLogsToLogger(
 
 				continue
 			}
-
 			m.Log(logger)
+
+			var ts int64
+			for _, stream := range m.Streams {
+				sts := stream.LatestTimestamp()
+				if sts > ts {
+					ts = sts
+				}
+			}
```

Comment:
I would argue you technically need to look ad the droppedEntries as well, but I don't think it matters all that much as that basically means we got *a lot of logs* and we loki couldn't send them all (or we couldn't read fast enough) and so even if we look at them and choose a different timestamp we will still probably be dropping a lot of logs, so I think just checking the Streams is fine for now

Reply:
Should I consider adding a comment regarding it?

Reply:
yeah, let's add a comment